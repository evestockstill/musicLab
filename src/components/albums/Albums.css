@import '../App.css';

.album {
  background: $black;
  border-radius: 1rem;
  padding: 0.6rem;
  height: 25rem;
  width: 100%;
  margin: 5rem 0.5;
  text-decoration: none;
  overflow: hidden;

  .record_image {
    width: 100%;
    border-top-left-radius: 1rem;
    border-top-right-radius: 1rem;
    display: block;
    padding: 1rem 0.6rem;
    text-decoration: none;
  }

  img {
    height: 20rem;
    width: auto;
    padding-bottom: 0.3rem;
    border-bottom: 4px solid $orange;
    background-color: black;
  }
  .img_container {
    position: relative;
    width: auto;
    border: 3px solid black;
    border-top-left-radius: 1rem;
    border-top-right-radius: 1rem;
  }
  .album_info {
    h3 {
      text-transform: uppercase;
      color: $green;
      background-color: $black2;
      text-decoration: none;
      text-align: center;
    }
  }
}
.Link {
  color: inherit;
  text-decoration: inherit;
  padding: 0.8rem 0.3rem 1rem 0.3rem;

  h3 {
    font-size: 1rem;
    overflow: hidden;
    padding: 0.8rem 0.3rem 1rem 0.3rem;
  }
}
.AlbumTitle {
  font-family: 'Permanent Marker', cursive;
  font-size: 3.5rem;
  text-align: center;
  margin: 1.5rem 1rem 0.2rem 1rem;
}
.ul {
  width: 70vw;
  margin: 3.4rem auto;
  padding: 1rem 0;
}
@media screen and (min-width: 550px) {
  .ul {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 2.5rem;
  }
}
@media screen and (min-width: 780px) {
  .ul {
    width: 85vw;
    grid-template-columns: repeat(3, 1fr);
  }
}
@media screen and (min-width: 1040px) {
  .ul {
    grid-template-columns: repeat(4, 1fr);
  }
}
@media screen and (min-width: 1280px) {
  .ul {
    grid-template-columns: repeat(5, 1fr);
  }
}
@media screen and (min-width: 1640px) {
  .ul {
    grid-template-columns: repeat(6, 1fr);
  }
}
.spinner {
  width: 40px;
  height: 40px;
  position: relative;
  margin: 100px auto;
}
.bounce1,
.bounce2 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: #333;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-animation: sk-bounce 2s infinite ease-in-out;
  animation: sk-bounce 2s infinite ease-in-out;
}
.bounce2 {
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}
@-webkit-keyframes sk-bounce {
  0%,
  100% {
    -webkit-transform: scale(0);
  }
  50% {
    -webkit-transform: scale(1);
  }
}
@keyframes sk-bounce {
  0%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  50% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
.error {
  display: flex;
  justify-content: center;
  font-size: 3rem;
  padding: 3rem;
  margin-top: 4rem;
}
