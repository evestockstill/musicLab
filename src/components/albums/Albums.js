import React from 'react';
import PropTypes from 'prop-types';
import { useAlbums } from '../../hooks/albums';
import { Link } from 'react-router-dom';

const Albums = ({ match }) => {
  const { albums, count } = useAlbums(match.params.id);
  
  if(!count) return <p>This artist has no albums.</p>;
  const albumsList = albums.map(album => {
    return (
      <Link
        key={album.id}
        to={`/tracks/${match.params.artistName}/${album.title}/${album.id}`}
      >
        <li>{album.title}</li>
      </Link>
    );
  });

  return (

    <ul>{albumsList} </ul>
  );
};

Albums.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
      artistName: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Albums;
