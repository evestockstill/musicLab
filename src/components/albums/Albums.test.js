import React from 'react';
import { shallow } from 'enzyme';
import Albums from './Albums';

describe('Albums component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Albums match={{ params: { id: '1' }
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
