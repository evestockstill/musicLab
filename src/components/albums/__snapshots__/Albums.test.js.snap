// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums component matches a snapshot 1`] = `
<p>
  This artist has no albums.
</p>
`;
